app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java

```diff
@@ -25,38 +25,100 @@
 import android.support.v7.widget.RecyclerView;
 import android.text.SpannableString;
 import android.text.style.StyleSpan;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.Menu;
 import android.view.MenuInflater;
 import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
+import android.widget.Toast;
 
 import com.g11x.checklistapp.data.Checklist;
-import com.g11x.checklistapp.data.ChecklistManager;
+import com.g11x.checklistapp.data.ChecklistItem;
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class ChecklistActivity extends AppCompatActivity {
 
-  private ChecklistAdapter adapter;
+  private ChecklistAdapter checklistAdapter;
+  private DatabaseReference checklistRef;
+  private final List<ChecklistItem> checklistItems = new ArrayList<>();
+  private ChildEventListener childEventListener;
+
+  private void addChildListeners(DatabaseReference reference) {
+    childEventListener = new ChildEventListener() {
+      @Override
+      public void onChildAdded(DataSnapshot dataSnapshot, String s) {
+        // Parse all current children and store locally, for now.
+        ChecklistItem item = dataSnapshot.getValue(ChecklistItem.class);
+        int originalSize = checklistItems.size();
+        checklistItems.add(item);
+        checklistAdapter.notifyDataSetChanged();
+      }
+
+      @Override
+      public void onChildChanged(DataSnapshot dataSnapshot, String s) {
+        // Do nothing, for now.  Should change list of items.
+      }
+
+      @Override
+      public void onChildRemoved(DataSnapshot dataSnapshot) {
+        // Do nothing, for now.  Should change list of items.
+      }
+
+      @Override
+      public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+        // Do nothing, for now.  Should re-order list of items.
+      }
+
+      @Override
+      public void onCancelled(DatabaseError databaseError) {
+        Log.e("Firebase", "Could not read data from remote database.");
+        Toast.makeText(getApplicationContext(), "Failed database read", Toast.LENGTH_SHORT);
```

Comment:
I think [Snackbars are slightly recommended](https://material.google.com/components/snackbars-toasts.html), and we're [using them elsewhere](https://github.com/g11x/checklistapp/blob/e56d35777e5368278b398541f641927c478513b9/app/src/main/java/com/g11x/checklistapp/ImportantInformationActivity.java#L76) currently.